storage/include/storage/storage_metrics.h

```diff
@@ -4,123 +4,41 @@
-    for (const auto& pair : ::rocksdb::TickersNameMap) {
-      if (std::find(tickers.begin(), tickers.end(), pair.first) != tickers.end()) {
-        auto metric_suffix = pair.second;
-        std::replace(metric_suffix.begin(), metric_suffix.end(), '.', '_');
-        active_tickers_.emplace(pair.first, rocksdb_comp_.RegisterAtomicGauge("storage_" + metric_suffix, 0));
-      }
-    }
-    rocksdb_comp_.Register();
-  }
 
-  /*
-   * For now, we have a hardcoded default metrics configuration list.
-   * In the future we will add a rocksdb configuration file to enable flexibility.
-   */
-  RocksDbStorageMetrics()
-      : RocksDbStorageMetrics({::rocksdb::Tickers::NUMBER_KEYS_WRITTEN,
-                               ::rocksdb::Tickers::NUMBER_KEYS_READ,
-                               ::rocksdb::Tickers::BYTES_WRITTEN,
-                               ::rocksdb::Tickers::BYTES_READ,
-                               ::rocksdb::Tickers::COMPACT_READ_BYTES,
-                               ::rocksdb::Tickers::COMPACT_WRITE_BYTES,
-                               ::rocksdb::Tickers::FLUSH_WRITE_BYTES,
-                               ::rocksdb::Tickers::STALL_MICROS,
-                               ::rocksdb::Tickers::BLOCK_CACHE_MISS,
-                               ::rocksdb::Tickers::BLOCK_CACHE_HIT,
-                               ::rocksdb::Tickers::BLOOM_FILTER_PREFIX_USEFUL,
-                               ::rocksdb::Tickers::BLOOM_FILTER_FULL_POSITIVE,
-                               ::rocksdb::Tickers::BLOOM_FILTER_FULL_TRUE_POSITIVE}) {}
+ protected:
+  // protected constructor to avoid creation of objects of this class
+  StorageMetricsBase(concordMetrics::Component metricsComponent, const size_t updateMetricsIntervalMsec)
+      : metrics_(metricsComponent),
+        update_metrics_(nullptr),
+        update_metrics_interval_msec_(updateMetricsIntervalMsec) {}
 
-  ~RocksDbStorageMetrics() { update_metrics_.reset(); }
-  void setAggregator(std::shared_ptr<concordMetrics::Aggregator> aggregator) {
-    rocksdb_comp_.SetAggregator(aggregator);
-  }
+  virtual ~StorageMetricsBase() { update_metrics_.reset(); }
 
-  void setMetricsDataSources(std::shared_ptr<::rocksdb::SstFileManager> sourceSstFm,
-                             std::shared_ptr<::rocksdb::Statistics> sourceStatistics) {
-    sstFm = sourceSstFm;
-    statistics = sourceStatistics;
-    update_metrics_ = std::make_unique<concord::util::PeriodicCall>([this]() { updateMetrics(); }, 100);
-  }
+  // the main func the derived class should implement
+  virtual void updateMetrics() = 0;
 
-  void updateMetrics() {
-    if (!sstFm || !statistics) return;
-    for (auto& pair : active_tickers_) {
-      pair.second.Get().Set(statistics->getTickerCount(pair.first));
-    }
-    total_db_disk_size_.Get().Set(sstFm->GetTotalSize());
-    rocksdb_comp_.UpdateAggregator();
-  }
+  // periodic update of metrics:
+  std::unique_ptr<concord::util::PeriodicCall> update_metrics_;
```

Comment:
Why update_metrics_ is her? It's initialized with nullptr (redundant btw) and never used.

Reply:
it's for the use of derived classes. this class can't be instantiated (ctor is protected)